pkg/didcomm/protocol/didexchange/states.go

```diff
@@ -276,7 +278,11 @@ func (ctx *context) handleInboundInvitation(invitation *Invitation, thid string)
 		return nil, err
 	}
 	// choose the first public key
-	sendVerKey := string(newDidDoc.PublicKey[0].Value)
+	pubKey, err := getPublicKey(newDidDoc, supportedPublicKeyType)
```

Comment:
Coverage decreased in all files due to this new error condition. You should add test for no key found scenario. You can do it by passing in mock did creator that returns did without public keys or with wrong key types.